Tasks/UsePythonV1/usepythonversion.ts

```diff
@@ -0,0 +1,146 @@
+//      <sys.prefix>/bin (by default /usr/local/bin, but not on hosted agents -- see the `else`)
+//      (--user) ~/.local/bin
+//   Windows:
+//      <Python installation dir>\Scripts
+//      (--user) %APPDATA%\Python\PythonXY\Scripts
+// See https://docs.python.org/3/library/sysconfig.html
+
+function binDir(installDir: string, platform: Platform): string {
+    if (platform === Platform.Windows) {
+        return path.join(installDir, 'Scripts');
+    } else {
+        return path.join(installDir, 'bin');
+    }
+}
+
+function pypyNotFoundError(majorVersion: 2 | 3) {
+    throw new Error([
+        task.loc('PyPyNotFound', majorVersion),
+        // 'Python' is intentional here
+        task.loc('ToolNotFoundMicrosoftHosted', 'Python', 'https://aka.ms/hosted-agent-software'),
+        task.loc('ToolNotFoundSelfHosted', 'Python', 'https://go.microsoft.com/fwlink/?linkid=871498')
+    ].join(os.EOL));
+}
+
+// Note on the tool cache layout for PyPy:
+// PyPy has its own versioning scheme that doesn't follow the Python versioning scheme.
+// A particular version of PyPy may contain one or more versions of the Python interpreter.
+// For example, PyPy 7.0 contains Python 2.7, 3.5, and 3.6-alpha.
+// We only care about the Python version, so we don't use the PyPy version for the tool cache.
+
+function usePyPy(majorVersion: 2 | 3, parameters: TaskParameters, platform: Platform): void {
+    const findPyPy = tool.findLocalTool.bind(undefined, 'PyPy', majorVersion.toString());
+    let installDir: string | null = findPyPy(parameters.architecture);
+
+    if (!installDir && platform === Platform.Windows) {
+        // PyPy only precompiles binaries for x86, but the architecture parameter defaults to x64.
+        // On Hosted VS2017, we only install an x86 version.
+        // Fall back to x86.
+        installDir = findPyPy('x86');
+    }
+
+    if (!installDir) {
+        // PyPy not installed in $(Agent.ToolsDirectory)
+        throw pypyNotFoundError(majorVersion);
+    }
+
+    // For PyPy, Windows uses 'bin', not 'Scripts'.
+    const _binDir = path.join(installDir, 'bin');
+
+    // On Linux and macOS, the Python interpreter is in 'bin'.
+    // On Windows, it is in the installation root.
+    const pythonLocation = platform === Platform.Windows ? installDir : _binDir;
+    task.setVariable('pythonLocation', pythonLocation);
+
+    // Add to path
+    toolUtil.prependPathSafe(installDir);
+    toolUtil.prependPathSafe(_binDir);
+}
+
+async function useCpythonVersion(parameters: Readonly<TaskParameters>, platform: Platform): Promise<void> {
+    const desugaredversion = desugarDevVersion(parameters.version);
+    const semanticversion: string = pythonVersionToSemantic(desugaredversion);
+    task.debug(`Semantic version spec of ${parameters.version} is ${semanticversion}`);
+
+    const installDir: string | null = tool.findLocalTool('Python', semanticversion, parameters.architecture);
+
+    console.log('INSTALL DIR - ' + installDir);
```

Comment:
I believe `findLocalTool` will already [print out a message](https://buildcanary.visualstudio.com/CanaryBuilds/_build/results?buildId=275651&view=logs&jobId=5f32d555-91de-5b47-aa96-4f80d9326a6b&taskId=0105a796-3810-57c6-c688-e428f3d4ee97&lineStart=9&lineEnd=10&colStart=1&colEnd=1) like "Found tool in cache: Python 3.7.2 x64"

If we want to log something for the user here, I think we should add more detail, and it should be localized.